pkg/webui/components/pagination/index.js

```diff
@@ -30,46 +43,112 @@ const Pagination = ({
   marginPagesDisplayed,
   hideIfOnlyOnePage,
   pageCount,
+  pageSize: propPageSize,
+  setPageSize,
   ...rest
 }) => {
+  const [selectedPage, setSelectedPage] = useState(forcePage)
+  const [pageSize, setQueryPageSize] = useQueryState('page-size', propPageSize)
```

Comment:
It's important to validate the page size as well and make sure that only values as provided in the select box are allowed.